text/0000-fabric-private-chaincode-1.0.md

```diff
@@ -0,0 +1,720 @@
+Hence, they can easily be reused.
+We plan to explore other TEE platforms such as AMD SEV in the future.
+
+## Fabric Touchpoints
+
+FPC 1.0 does ***not*** require any changes/modifications to Fabric (Peer).
+We recommend reading the chaincode development and deployment sections to know more about the private chaincode coding language and the use of Fabric's external builder and launcher capabilities.
+
+## Requirements
+
+- FPC Chaincode must be written in C/C++ using our FPC SDK. More language support is planned.
+- FPC relies on the External Builder and Launcher feature of Fabric.
+  <!-- comment out below as we haven't introduced registry yet and, at least so far, we have put the registration under the cover of cli
+   FPC's attestation infrastructure requires the installation of an FPC Registry chaincode per channel.
+  -->
+- FPC requires that the Endorsing Peers run on an Intel&reg; x86 machine with the SGX feature enabled. We plan to add support for other Trusted Execution Environments in future releases.
+
+## Fabric Features Not (Yet) Supported
+
+In order to focus the development resources on the core components of FPC, the FPC 1.0 initially excludes certain Fabric features, which will be added in the future.
+
+- Multiple implementations for a single chaincode.
+This feature is supported in Fabric v2 and gives organizations the freedom to implement and package their own chaincode.
+It allows for different chaincode implementations as long as changes to the ledger state implement the same agreed upon state machine, i.e., the application integrity is ensured.
+
+    FPC Chaincodes have a stronger requirement: Not only must we be assured of the application integrity but we also require that all information flows be controlled to meet our confidentiality requirements.   As the execution during endorsement is unobservable by other organizations, they will require the assurance that any chaincode getting access to the state decryption keys, and hence sensitive information, will never leak unintended information.  Therefore, the implementation of a chaincode must allow for public examination for (lack of) potential leaks of confidential data.
+    Only then clients can establish trust in how the chaincode executable treats their sensitive data.
+
+    For a given chaincode, the FPC 1.0 currently supports only a single active implementation.
+    Most importantly, the version field of the chaincode definition precisely identifies the chaincode's binary executable.
+
+- Multiple key/value pairs and composite keys as well as secure access to MSP identities via `getCreator` will be supported once below [Rollback-Protection Extension](#rollback-protection-extension) is added.
+- Arbitrary endorsement policies
+- State-based endorsement
+- Chaincode-to-chaincode invocations (cc2cc)
+- Private Collections
+- Complex (CouchDB) queries like range queries
```

Comment:
Actually even goleveldb supports range queries. CouchDB only adds JSON queries (which cannot be used in write transactions anyways).